builtin/audit/syslog/backend.go

```diff
@@ -75,125 +72,61 @@ func Factory(_ context.Context, conf *audit.BackendConfig, useEventLogger bool,
 		return nil, fmt.Errorf("%s: cannot configure a fallback device with a filter: %w", op, event.ErrInvalidParameter)
 	}
 
-	cfg, err := formatterConfig(conf.Config)
-	if err != nil {
-		return nil, fmt.Errorf("%s: failed to create formatter config: %w", op, err)
-	}
-
 	// Get the logger
 	logger, err := gsyslog.NewLogger(gsyslog.LOG_INFO, facility, tag)
 	if err != nil {
 		return nil, fmt.Errorf("%s: cannot create logger: %w", op, err)
 	}
 
 	b := &Backend{
-		fallback:     fallback,
-		formatConfig: cfg,
-		logger:       logger,
-		name:         conf.MountPath,
-		saltConfig:   conf.SaltConfig,
-		saltView:     conf.SaltView,
+		fallback:   fallback,
+		logger:     logger,
+		name:       conf.MountPath,
+		saltConfig: conf.SaltConfig,
+		saltView:   conf.SaltView,
+		nodeIDList: []eventlogger.NodeID{},
+		nodeMap:    make(map[eventlogger.NodeID]eventlogger.Node),
 	}
 
-	// Configure the formatter for either case.
-	f, err := audit.NewEntryFormatter(b.formatConfig, b, audit.WithHeaderFormatter(headersConfig), audit.WithPrefix(conf.Config["prefix"]))
+	err = b.configureFilterNode(conf.Config["filter"])
 	if err != nil {
-		return nil, fmt.Errorf("%s: error creating formatter: %w", op, err)
+		return nil, fmt.Errorf("%s: error configuring filter node: %w", op, err)
 	}
 
-	var w audit.Writer
-	switch b.formatConfig.RequiredFormat {
-	case audit.JSONFormat:
-		w = &audit.JSONWriter{Prefix: conf.Config["prefix"]}
-	case audit.JSONxFormat:
-		w = &audit.JSONxWriter{Prefix: conf.Config["prefix"]}
-	}
-
-	fw, err := audit.NewEntryFormatterWriter(b.formatConfig, f, w)
+	cfg, err := formatterConfig(conf.Config)
 	if err != nil {
-		return nil, fmt.Errorf("%s: error creating formatter writer: %w", op, err)
+		return nil, fmt.Errorf("%s: failed to create formatter config: %w", op, err)
 	}
 
-	b.formatter = fw
-
-	if useEventLogger {
```

Comment:
This is mostly out of curiosity, but wouldn't we need to keep this section and execute it every time now? Or is this only relevant with the two logging modes in tandem?

Reply:
Yea, the stuff that happened inside that `useEventLogger` is still around, but GH is having a really hard time seeing what moved/changed due to the if/else blocks going, some unindenting and what not. 

Factory methods now follow this vague outline:

* Parse the params supplied and validate we have all the required (and optional) bits for a particular backend
* Create the `&Backend{}` object with fields initialised
* Hand off to `configure___` style receivers on the `Backend` (`configureFilterNode`, `configureFormatterNode`, `configureSinkNode`) that handle configuring filter/formatter/sink nodes

Does that help at all? 😄

Reply:
Got it! Thanks! I figured it'd be something like that. Appreciate it :D